src/schema/users.ts

```diff
@@ -1866,6 +1880,71 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
         weekStart,
       };
     },
+    recoverStreak: async (
+      _,
+      __,
+      ctx: AuthContext,
+      info,
+    ): Promise<GQLUserStreak> => {
+      const { userId } = ctx;
+
+      const oldStreakLength = await getRestoreStreakCache({ userId });
+      if (!oldStreakLength) {
+        throw new ValidationError('No streak to recover');
+      }
+
+      const streak = await getUserStreakQuery(userId, ctx, info);
+      const hasNoStreakOrCurrentIsGreaterThanOne =
+        !streak || streak.current > 1;
+      if (hasNoStreakOrCurrentIsGreaterThanOne) {
+        throw new ValidationError('Time to recover streak has passed');
+      }
+
+      const [user, lastUserRecoverAction] = await Promise.all([
+        ctx.con.getRepository(User).findOneByOrFail({ id: userId }),
+        await ctx.con.getRepository(UserStreakAction).findOneBy({
+          userId,
+          type: UserStreakActionType.Recover,
+        }),
+      ]);
+      const isFirstRecover = !lastUserRecoverAction;
+      const recoverCost = isFirstRecover ? 0 : streakRecoverCost;
+      const userCanAfford = user.reputation >= recoverCost;
```

Comment:
Can't we just use `reputationReasonAmount` here?

Reply:
Its fine this way around if you prefer just need to replace `0` with `reputationReasonAmount`